pom.xml

```diff
@@ -581,6 +588,7 @@
                     </resourceExcludes>
                     <excludes>
                         **/target/generated-test-sources/**,
+                        org/apache/skywalking/apm/agent/core/logging/core/generated/LoggingBenchmark*.java,
```

Comment:
Could you share which part generates these classes? 

And please move this to the last one.

Reply:
Fixed.

These synthetic codes are generated by the "jmh-annotation-processor" according to the [document](https://openjdk.java.net/projects/code-tools/jmh/).

In the past, this dependency is not included even if there is another benchmark `LinkedArrayBenchmark`. I guess this was the trick.

Reply:
Could you highlight which parts of documents talking about this code generation?

Reply:
`LinkedArrayBenchmark` doesn' generate codes, as you can see from the `excludes` list, nothing about DataCarrier packages.

Reply:
In the "Basic Considerations" section, it reads

> In all cases, the key to using JMH is enabling the annotation- or bytecode-processors to generate the synthetic benchmark code.

Reply:
> LinkedArrayBenchmark doesn' generate codes, as you can see from the excludes list, nothing about DataCarrier packages.

Could you check why this? Even now, it doesn't generate new codes. I prefer we don't need to generate anything.
Less dependency(even test scope only), less maintenance.

Reply:
Also, without adding this dependency, `LinkedArrayBenchmark` still is able to run in the IntelliJ, what is the difference or do I miss anything?

Reply:
> Also, without adding this dependency, `LinkedArrayBenchmark` still is able to run in the IntelliJ, what is the difference or do I miss anything?

I've tried to run this benchmark, but I failed. An error is reported,

```
Exception in thread "main" java.lang.RuntimeException: ERROR: Unable to find the resource: /META-INF/BenchmarkList
	at org.openjdk.jmh.runner.AbstractResourceReader.getReaders(AbstractResourceReader.java:98)
	at org.openjdk.jmh.runner.BenchmarkList.find(BenchmarkList.java:122)
	at org.openjdk.jmh.runner.Runner.internalRun(Runner.java:263)
	at org.openjdk.jmh.runner.Runner.run(Runner.java:209)
	at org.apache.skywalking.apm.commons.datacarrier.LinkedArrayBenchmark.main(LinkedArrayBenchmark.java:176)
```

How can you run this benchmark in IDEA?

Reply:
Probably you have some special setup with your IDE? Actually I found some issues related to the "jmh-generator-annprocess",

https://github.com/artyushov/idea-jmh-plugin/issues/13
https://github.com/undertow-io/undertow/pull/833

Reply:
@kezhenxu94 Can you also kindly make a cross-check?

Reply:
> @kezhenxu94 Can you also kindly make a cross-check?

The same error as yours

Reply:
But why that benchmark doesn't generate anything? That is why I am asking the question.

Reply:
> But why that benchmark doesn't generate anything? That is why I am asking the question.

I am not clear about what happened in the past. I guess it was a trick to avoid the style-check issue. But I am sure the dependency should have been declared there.

I am still curious why you can run it seamlessly?

Reply:
> I am still curious why you can run it seamlessly?

I am not sure either. The dependency seems not an issue. I am just checking the inconsistent issue, two benchmarks having different exclusion configuration concerns me.

> As I understand, this LOC should help

Then why your benchmark isn't excluded?